applications/GeoMechanicsApplication/tests/cpp_tests/test_thermal_dispersion_law.cpp

```diff
@@ -0,0 +1,115 @@
+    Geometry<Node> geometry;
+
+    const Matrix thermal_dispersion_matrix =
+        geo_thermal_dispersion2d_law.CalculateThermalDispersionMatrix(
+            *cond_prop, info, geometry);
+
+    Matrix expected_solution = ZeroMatrix(2, 2);
+    expected_solution(0, 0) = 1125.0;
+    expected_solution(0, 1) = 1000.0;
+    expected_solution(1, 0) = expected_solution(0, 1);
+    expected_solution(1, 1) = 975.0;
+
+    constexpr double tolerance{1.0e-6};
+
+    for (unsigned int i = 0; i < thermal_dispersion_matrix.size1(); ++i) {
+        for (unsigned int j = 0; j < thermal_dispersion_matrix.size2(); ++j) {
+            KRATOS_EXPECT_NEAR(thermal_dispersion_matrix(i, j),
+                               expected_solution(i, j), tolerance);
+        }
+    }
+}
+
+KRATOS_TEST_CASE_IN_SUITE(CalculateThermalDispersionMatrix3D, KratosGeoMechanicsFastSuite)
+{
+    Model current_model;
+    auto& r_model_part = current_model.CreateModelPart("ModelPart", 1);
+
+    auto cond_prop = r_model_part.CreateNewProperties(0);
+    cond_prop->SetValue(POROSITY, 0.5);
+    cond_prop->SetValue(SATURATED_SATURATION, 0.75);
+    cond_prop->SetValue(RETENTION_LAW, "SaturatedLaw");
+    cond_prop->SetValue(THERMAL_CONDUCTIVITY_WATER, 800.0);
+    cond_prop->SetValue(THERMAL_CONDUCTIVITY_SOLID_XX, 1000.0);
+    cond_prop->SetValue(THERMAL_CONDUCTIVITY_SOLID_XY, 2000.0);
+    cond_prop->SetValue(THERMAL_CONDUCTIVITY_SOLID_YY, 1000.0);
+    cond_prop->SetValue(THERMAL_CONDUCTIVITY_SOLID_XZ, 1000.0);
+    cond_prop->SetValue(THERMAL_CONDUCTIVITY_SOLID_YZ, 2000.0);
+    cond_prop->SetValue(THERMAL_CONDUCTIVITY_SOLID_ZZ, 500.0);
+
+    const SizeType dimension = 3;
+    GeoThermalDispersionLaw geo_thermal_dispersion_2d_law(dimension);
+    ProcessInfo info;
+    Geometry<Node> geometry;
+
+    const Matrix thermal_dispersion_matrix =
+        geo_thermal_dispersion_2d_law.CalculateThermalDispersionMatrix(
+            *cond_prop, info, geometry);
+
+    Matrix expected_solution = ZeroMatrix(3, 3);
+    expected_solution(0, 0) = 800.0;
+    expected_solution(0, 1) = 1000.0;
+    expected_solution(0, 2) = 500.0;
+    expected_solution(1, 0) = expected_solution(0, 1);
+    expected_solution(1, 1) = expected_solution(0, 0);
+    expected_solution(1, 2) = 1000.0;
+    expected_solution(2, 0) = expected_solution(0, 2);
+    expected_solution(2, 1) = expected_solution(1, 2);
+    expected_solution(2, 2) = 550.0;
+
+    constexpr double tolerance{1.0e-6};
+
+    for (unsigned int i = 0; i < thermal_dispersion_matrix.size1(); ++i) {
+        for (unsigned int j = 0; j < thermal_dispersion_matrix.size2(); ++j) {
+            KRATOS_EXPECT_NEAR(thermal_dispersion_matrix(i, j),
+                               expected_solution(i, j), tolerance);
+        }
+    }
+}
+
+KRATOS_TEST_CASE_IN_SUITE(GetWorkingSpaceDimension_ReturnsCorrectValue, KratosGeoMechanicsFastSuite)
+{
+    const SizeType dimension = 3;
+    GeoThermalDispersionLaw geo_thermal_dispersion_2D_law(dimension);
```

Comment:
I would also rename this local variable for consistency:
```suggestion
    GeoThermalDispersionLaw geo_thermal_dispersion_3D_law(dimension);
```
And perhaps also add a check (in this test case) for the 2D case.

Reply:
corrected